integrationTests/chainSimulator/bridge/executeOrDepositFail_test.go

```diff
@@ -0,0 +1,370 @@
+		Type:       core.NonFungible,
+	})
+
+	testExecuteOperationNotAllowedToMintFungibleContractNotWhitelisted(t, prefix, bridgedInTokens)
+}
+
+func TestChainSimulator_ExecuteOperationNotAllowedToMintSemiFungibleContractNotWhitelisted(t *testing.T) {
+	prefix := "sov3"
+	bridgedInTokens := make([]chainSim.ArgsDepositToken, 0)
+	bridgedInTokens = append(bridgedInTokens, chainSim.ArgsDepositToken{
+		Identifier: prefix + "-SOVT-5d8f56",
+		Nonce:      3,
+		Amount:     big.NewInt(15),
+		Type:       core.SemiFungible,
+	})
+
+	testExecuteOperationNotAllowedToMintFungibleContractNotWhitelisted(t, prefix, bridgedInTokens)
+}
+
+// Test flow:
+// - deploy sovereign bridge contracts on the main chain
+// - whitelist WRONG bridge esdt safe contract not allowing to burn/mint cross chain esdt tokens
+// - executeOperation token with prefix should not be allowed
+func testExecuteOperationNotAllowedToMintFungibleContractNotWhitelisted(
+	t *testing.T,
+	prefix string,
+	bridgedInTokens []chainSim.ArgsDepositToken,
+) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	whiteListedAddress := "erd1qqqqqqqqqqqqqpgqcw92wj0huvaghg4aeuykknp7hstmrmhudz3shjdhtt"
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   true,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              1,
+		GenesisTimestamp:         time.Now().Unix(),
+		RoundDurationInMillis:    uint64(6000),
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  0,
+		NumNodesWaitingListShard: 0,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.GeneralConfig.VirtualMachine.Execution.TransferAndExecuteByUserAddresses = []string{whiteListedAddress}
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocksUntilEpochIsReached(3)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(0)
+
+	// Deploy bridge setup
+	initialAddress := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	argsEsdtSafe := ArgsEsdtSafe{
+		ChainPrefix:       prefix,
+		IssuePaymentToken: "ABC-123456",
+	}
+	bridgeData := deployBridgeSetup(t, cs, initialAddress, simpleEsdtSafeWasmPath, argsEsdtSafe, feeMarketWasmPath)
+
+	// esdt-safe address generated is NOT whitelisted
+	esdtSafeEncoded, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Encode(bridgeData.ESDTSafeAddress)
+	require.NotEqual(t, whiteListedAddress, esdtSafeEncoded)
+
+	wallet, err := cs.GenerateAndMintWalletAddress(0, chainSim.InitialAmount)
+	require.Nil(t, err)
+
+	// execute operation, bridge in token with prefix
+	// expecting not allowed to mint because ESDT-safe contract is not whitelisted
```

Comment:
Same question as above

Reply:
The error is `signalError` event with `action is not allowed` and is checked. This is what VM is returning. Do we expect something else?